std/socket.d

```diff
@@ -1663,6 +1663,22 @@ public:
     }
 
     /**
+     * Compares with another InternetAddress of same type for equality
+     * Returns: true if the InternetAddresses share the same address and
+     * port number.
+     * Examples:
+     * --------------
+     * InternetAddress addr1,addr2;
+     * if (addr1 == addr2) { }
```

Comment:
I tested it and it seems to work.  http://dlang.org/operatoroverloading.html


> If a and b are both class objects, then the expression is rewritten as:
`.object.opEquals(a, b)`
and that function is implemented as:
```D
bool opEquals(Object a, Object b)
{
    if (a is b) return true;
    if (a is null || b is null) return false; // NULL HANDLED HERE
    if (typeid(a) == typeid(b)) return a.opEquals(b);
    return a.opEquals(b) && b.opEquals(a);
}
```